### DISCUSSION TITLE
Augmentation - Hover event_type changed?

### DISCUSSION THREAD

**Discussion by chad63e** [anvilistas/anvil-extras#319] [Q&A] [answered]
It looks like the anvil_extras “augment” function is now returning a different response for the “hover” event.

Per the documentation, it states event_type “enter” is what to look for. Now it is “mouseover” and “mouseout”? If this is now going to be changed I will have to go through ALL of my code and replace this!

Please confirm what we are to expect?

Previous:

```
from anvil_extras import augment
augment.set_event_handler(self.link, 'hover', self.link_hover)

def link_hover(self, **event_args):
  if 'enter' in event_args['event_type']:
    self.link.text = 'hover'
  else:
    self.link.text = 'hover_out'
Now:
```

```
from anvil_extras import augment
augment.set_event_handler(self.link, 'hover', self.link_hover)

def link_hover(self, **event_args):
  if 'mouseover' in event_args['event_type']:
    self.link.text = 'hover'
  else:
    self.link.text = 'hover_out'
```

Also, sometimes it is: “mouseleave”, or “mouseenter”. This does not seem dependable?

**Comment by s-cork** [comment]
Not sure. 
The code hasn't changed for hover since it was written. 

You could see if it's different for links and buttons. 

I've timed out for today but I can look more at this tomorrow. 

Please can you provide a clone link and browser/OS information in the meantime. (In case it's browser specific)

**Comment by chad63e** [comment]
I've tried it in both Chrome and Brave and gotten the same results.

Here is a sample app showing how it randomly gives different event_types. Just scroll over the different component types multiple times and you will notice that the same component will give different results from time to time.

https://anvil.works/build#clone:6ZXLBLDOIK7FQULE=D3S4RIMDM7E7WITROO5RFMOQ

It's easily fixable with a little code change, but I'm concerned I will have to update all the code I've ever written!

This is what I am using to resolve the randomness:

```
def link_hover(self, **event_args):
    if event_args['event_type'] in ["enter", "mouseenter", "mouseover"]:
      self.toggle_highlight(True)
    else:
      self.toggle_highlight(False)
```

**Comment by s-cork** [comment]
Are you on windows?
On Mac I only get `mouseenter` and `mouseleave`
(which is the expected values)

the docs say to do `enter in event_args["event_type"]` which expects the value to be `mouseenter`

**Comment by chad63e** [comment]
I'm on windows

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
Is this new behaviour?
Or you've only just noticed?
since when have you noticed it?

**Comment by chad63e** [comment]
That's why I initially posted on the regular forum.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
What were the answers?

Is this new behaviour?
Or you've only just noticed?
since when have you noticed it?

**Comment by chad63e** [comment]
Sorry,

This is new behavior, and yes, it just started today after the anvil issue

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by s-cork** [reply]
Can you try again. There was an IDE bug we just fixed that may have solved this already.

**Comment by chad63e** [comment]
I'm out of the office now, but will see what it is returning tomorrow and let you know. Thanks

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by s-cork** [comment, marked as answer]
I've managed to reproduce if I reintroduce the beta bug that was fixed.
So this should now work as expected.

I've also implemented a workaround that will prevent this happening again see #319

**Comment by chad63e** [comment]
It looks like it is consistantly "mouseenter" or "mouseleave" now.  Thanks for the help Stu!

**Comment by chad63e** [comment]
Yes, it started right after the issue with anvil not loading nested forms today.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
